code/game/objects/items/weapons/batons.dm

```diff
@@ -39,27 +46,65 @@
 			user.take_organ_damage(force * 2)
 		return
 
-	if(user.a_intent == INTENT_HARM || isrobot(target)) // Lethal attack or it's a borg (can't knock them down!)
+	if(user.a_intent == INTENT_HARM)
 		return ..()
-	else if(!on_cooldown) // Non-lethal attack - knock them down
-		// Check for shield/countering
-		if(ishuman(target))
-			var/mob/living/carbon/human/H = target
-			if(H.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK))
-				return
-			if(check_martial_counter(H, user))
-				return
-		// Visuals and sound
-		user.do_attack_animation(target)
-		playsound(target, 'sound/effects/woodhit.ogg', 75, TRUE, -1)
-		add_attack_logs(user, target, "Stunned with [src]")
-		target.visible_message("<span class='danger'>[user] has knocked down [target] with \the [src]!</span>", \
-							   "<span class='userdanger'>[user] has knocked down [target] with \the [src]!</span>")
-		// Hit 'em
-		target.LAssailant = iscarbon(user) ? user : null
-		target.Weaken(3)
-		on_cooldown = TRUE
-		addtimer(CALLBACK(src, .proc/cooldown_finished), BATON_STUN_COOLDOWN)
+	if(on_cooldown)
+		return
+	if(issilicon(target))
+		if(affect_silicon)
+			stun_silicon(target, user)
+		else
+			return ..()
+	else
+		stun(target, user)
+
+/**
+  * Called when a silicon target is about to be hit non-lethally.
+  *
+  * Arguments:
+  * * target - The silicon mob about to be hit
+  * * user - The attacking user
+  */
+/obj/item/melee/classic_baton/proc/stun_silicon(mob/living/silicon/target, mob/living/user)
```

Comment:
I'd rather have two separate procs in case overriding classes have specific behaviour on silicon/non-silicon stunning. That said I will get rid of the duplicate code where possible.